core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/keystore/persistence/constant/PersistenceManagerConstants.java

```diff
@@ -45,4 +47,51 @@ private RegistryResources() {
         public static final String TENANT_PUBKEY_RESOURCE = ROOT + "security/pub-key";
     }
 
+    public static class KeyStoreTableColumns {
+
+        private KeyStoreTableColumns() {
+
+        }
+
+        public static final String ID = "ID";
+        public static final String NAME = "NAME";
+        public static final String TYPE = "TYPE";
+        public static final String PROVIDER = "PROVIDER";
+        public static final String PASSWORD = "PASSWORD";
+        public static final String PRIVATE_KEY_ALIAS = "PRIVATE_KEY_ALIAS";
+        public static final String PRIVATE_KEY_PASS = "PRIVATE_KEY_PASS";
+        public static final String CONTENT = "CONTENT";
+        public static final String PUB_CERT_ID = "PUB_CERT_ID";
+        public static final String TENANT_ID = "TENANT_ID";
+        public static final String VERSION = "VERSION";
+        public static final String CREATED_AT = "CREATED_AT";
+        public static final String UPDATED_AT = "UPDATED_AT";
+    }
+
+    public static class SqlQueries {
+
+        private SqlQueries() {
+
+        }
+
+        public static final String ADD_KEY_STORE = "INSERT INTO KEY_STORE (NAME, TYPE, PROVIDER, PASSWORD, " +
+                "PRIVATE_KEY_ALIAS, PRIVATE_KEY_PASS, CONTENT, PUB_CERT_ID, TENANT_ID, VERSION, CREATED_AT, " +
+                "UPDATED_AT) VALUES (:NAME;, :TYPE;, :PROVIDER;, :PASSWORD;, :PRIVATE_KEY_ALIAS;, " +
+                ":PRIVATE_KEY_PASS;, :CONTENT;, :PUB_CERT_ID;, :TENANT_ID;, :VERSION;, :CREATED_AT;, :UPDATED_AT;);";
+        public static final String GET_KEY_STORE_BY_NAME = "SELECT NAME, TYPE, PROVIDER, PASSWORD, " +
+                "PRIVATE_KEY_ALIAS, PRIVATE_KEY_PASS, CONTENT, TENANT_ID, PUB_CERT_ID  FROM KEY_STORE WHERE " +
+                "NAME = :NAME; AND TENANT_ID = :TENANT_ID;";
+        public static final String GET_KEY_STORE_LAST_UPDATED_TIME =
+                "SELECT UPDATED_AT FROM KEY_STORE WHERE NAME = :NAME; AND TENANT_ID = :TENANT_ID;";
+        public static final String GET_ENCRYPTED_KEY_STORE_PASSWORD =
+                "SELECT PASSWORD FROM KEY_STORE WHERE NAME = :NAME; AND TENANT_ID = :TENANT_ID;";
+        public static final String GET_ENCRYPTED_PRIVATE_KEY_PASSWORD =
+                "SELECT PRIVATE_KEY_PASS FROM KEY_STORE WHERE NAME = :NAME; AND TENANT_ID = :TENANT_ID;";
+        public static final String GET_KEY_STORES = "SELECT * FROM KEY_STORE WHERE TENANT_ID = :TENANT_ID;";
+        public static final String GET_KEY_STORES_EXISTENCE = "SELECT ID FROM KEY_STORE WHERE TENANT_ID = :TENANT_ID;";
```

Comment:
Rather than quering the list of IDs, if we are to check the existence only, lets query for count